src/components/ObsEdit/BottomButtons.js

```diff
@@ -1,40 +1,115 @@
 // @flow
 
+import { useNavigation } from "@react-navigation/native";
 import classnames from "classnames";
 import {
   Button, StickyToolbar
 } from "components/SharedComponents";
 import { View } from "components/styledComponents";
-import { ObsEditContext } from "providers/contexts";
+import { RealmContext } from "providers/contexts";
 import type { Node } from "react";
-import React, { useContext, useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
+import Observation from "realmModels/Observation";
+import { writeExifToFile } from "sharedHelpers/parseExif";
+import uploadObservation from "sharedHelpers/uploadObservation";
 import useTranslation from "sharedHooks/useTranslation";
 
+import { log } from "../../../react-native-logs.config";
 import ImpreciseLocationSheet from "./Sheets/ImpreciseLocationSheet";
 import MissingEvidenceSheet from "./Sheets/MissingEvidenceSheet";
 
+const { useRealm } = RealmContext;
+
 const DESIRED_LOCATION_ACCURACY = 4000000;
 
 type Props = {
   passesEvidenceTest: boolean,
-  passesIdentificationTest: boolean
+  passesIdentificationTest: boolean,
+  observations: Array<Object>,
+  currentObservation: Object,
+  unsavedChanges: boolean,
+  currentObservationIndex: number,
+  cameraRollUris: Array<string>,
+  setCurrentObservationIndex: Function
 }
 
 const BottomButtons = ( {
   passesEvidenceTest,
-  passesIdentificationTest
+  passesIdentificationTest,
+  currentObservation,
+  unsavedChanges,
+  currentObservationIndex,
+  observations,
+  cameraRollUris,
+  setCurrentObservationIndex
 }: Props ): Node => {
+  const navigation = useNavigation( );
   const { t } = useTranslation( );
-  const {
-    setNextScreen,
-    currentObservation,
-    unsavedChanges,
-    loading
-  } = useContext( ObsEditContext );
+  const realm = useRealm( );
   const [showMissingEvidenceSheet, setShowMissingEvidenceSheet] = useState( false );
   const [showImpreciseLocationSheet, setShowImpreciseLocationSheet] = useState( false );
   const [allowUserToUpload, setAllowUserToUpload] = useState( false );
   const [buttonPressed, setButtonPressed] = useState( null );
+  const [loading, setLoading] = useState( false );
+
+  const logger = log.extend( "ObsEditBottomButtons" );
+
+  const ensureRealm = ( ) => {
+    if ( !realm ) {
+      throw new Error( "Gack, tried to save an observation without realm!" );
+    }
+  };
+
+  const writeExifToCameraRollPhotos = async exif => {
+    if ( !cameraRollUris || cameraRollUris.length === 0 || !currentObservation ) {
+      return;
+    }
+    // Update all photos taken via the app with the new fetched location.
+    cameraRollUris.forEach( uri => {
+      logger.info( "writeExifToCameraRollPhotos, writing exif for uri: ", uri );
+      writeExifToFile( uri, exif );
+    } );
+  };
+
+  const saveObservation = async observation => {
+    ensureRealm( );
```

Comment:
What happens when you don't do this?

Reply:
I think it's safe to remove. That was code we had in ObsEditProvider but doesn't seem necessary.